x/ecocredit/server/basket/util.go

```diff
@@ -0,0 +1,59 @@
+package basket
+
+import (
+	"context"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/basket/v1"
+	"github.com/regen-network/regen-ledger/types/math"
+)
+
+func (k Keeper) GetBasketBalanceMap(ctx context.Context) (map[uint64]math.Dec, error) {
+	itr, err := k.stateStore.BasketTable().List(ctx, api.BasketPrimaryKey{})
+	if err != nil {
+		return nil, err
+	}
+	defer itr.Close()
+
+	batchDenomToId := make(map[string]uint64)     // map of a batch denom to batch id
+	batchIdToBalance := make(map[uint64]math.Dec) // map of a basket batch_id to balance
+	for itr.Next() {
+		basket, err := itr.Value()
+		if err != nil {
+			return nil, err
+		}
+
+		bb, err := k.stateStore.BasketBalanceTable().Get(ctx, basket.Id, basket.BasketDenom)
```

Comment:
I'm a bit confused why we're getting the balance of the basket tokens here and not the balance of the credits within the basket. A basket denom is the bank denom that takes the form of `eco.dC.NCT` whereas a batch denom is the ecocredit denom that takes the form of `C01-00000000-00000000-001`. We should be looping through the batches within a basket to get the credit balance of each batch within the basket.